main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -631,6 +631,16 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       }
     }
 
+    def Select: Rule1[ParsedAst.Expression.Select] = {
+      def Rule: Rule1[ParsedAst.SelectRule] = rule {
+        atomic("case") ~ WS ~ Names.Variable ~ optWS ~ "=" ~ optWS ~ "<-" ~ optWS ~ Expression ~ optWS ~ atomic("=>") ~ optWS ~ Expression ~> ParsedAst.SelectRule
+      }
+
+      rule {
+        SP ~ atomic("select") ~ WS ~ "{" ~ optWS ~ oneOrMore(Rule).separatedBy(optWS) ~ optWS ~ "}" ~ SP ~> ParsedAst.Expression.Select
```

Comment:
- Why hard WS after select? Should `select{` not be allowed?
- I think rules should be separated by hard ws, otherwise you can write `case x <- c => 42case y <- c => 21`